modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/ValidationHelper.java

```diff
@@ -131,6 +146,10 @@ public void validateAPIEndpointRelationship(
 		}
 	}
 
+	private static final Collection<String>
+		_allowedUnmodifiableSystemObjectDefinitionNames = Arrays.asList(
```

Comment:
For searching purposes (like this), Set is faster:

```
Benchmark                 Mode   Cnt   Score   Error  Units
searchElementInArrayList     ss   15   0.014 ± 0.015   s/op
searchElementInHashSet       ss   15   ≈ 10⁻⁵          s/op


The result shows that searching for an element in a HashSet is faster than searching for an element in an ArrayList. This ascertains that a HashSet is more efficient in a scenario where we want to search for an element in a collection in a fast and efficient way.
```

https://www.baeldung.com/java-set-vs-list#3-contains

An usage in the Liferay portal:

https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/document-library-google-docs/document-library-google-docs/src/main/java/com/liferay/document/library/google/docs/internal/item/selector/criterion/DLFileItemSelectorCriterionCreationMenuRestriction.java#L38

Reply:
oh, that's a valuable information, thanks! 👍🏻